internal/commands/build.go

```diff
@@ -143,6 +145,13 @@ func buildCommandFlags(cmd *cobra.Command, buildFlags *BuildFlags, cfg config.Co
 	cmd.Flags().StringArrayVar(&buildFlags.EnvFiles, "env-file", []string{}, "Build-time environment variables file\nOne variable per line, of the form 'VAR=VALUE' or 'VAR'\nWhen using latter value-less form, value will be taken from current\n  environment at the time this command is executed\nNOTE: These are NOT available at image runtime.\"")
 	cmd.Flags().StringVar(&buildFlags.Network, "network", "", "Connect detect and build containers to network")
 	cmd.Flags().BoolVar(&buildFlags.Publish, "publish", false, "Publish to registry")
+	cmd.Flags().StringVar(&buildFlags.DockerHost, "docker-host", "inherit",
```

Comment:
Apologies, but I realize I missed one aspect of the workflow in my [earlier comment](https://github.com/buildpacks/pack/pull/988#pullrequestreview-564491641), that the previous behavior needs to be the new default (namely, when `--docker-host` is not provided, don't pass through `DOCKER_HOST` env vars into the container and always using a bind mount). Unfortunately, `inherit` isn't sufficient for a backward compatible default. 

I was able to fix this in your PR by setting the default value for the flag to `""` instead of `"inherit"`. I'd be happy with this change for now and if needs any further cleanup, I can do another pass when I add in the TLS behavior.